crates/turborepo-lib/src/run/summary/execution.rs

```diff
@@ -1,28 +1,164 @@
 use std::fmt;
 
-use chrono::{DateTime, Duration, Local};
-use serde::{ser::SerializeStruct, Serialize};
+use chrono::{DateTime, Duration, Local, SubsecRound};
+use serde::{ser::SerializeStruct, Serialize, Serializer};
 use tokio::sync::mpsc;
-use turbopath::AnchoredSystemPath;
+use tracing::log::warn;
+use turbopath::{AbsoluteSystemPathBuf, AnchoredSystemPath};
+use turborepo_ui::{color, BOLD, BOLD_GREEN, BOLD_RED, MAGENTA, UI};
 
-use crate::run::task_id::TaskId;
+use crate::run::{summary::task::TaskSummary, task_id::TaskId};
 
 // Just used to make changing the type that gets passed to the state management
 // thread easy
 type Message = Event;
 
+fn serialize_datetime<S: Serializer>(
+    date_time: &DateTime<Local>,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    serializer.serialize_i64(date_time.timestamp_millis())
+}
+
+fn serialize_optional_datetime<S: Serializer>(
+    date_time: &Option<DateTime<Local>>,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    let millis = date_time
+        .map(|dt| dt.timestamp_millis())
+        .unwrap_or_default();
+    serializer.serialize_i64(millis)
+}
+
 // Should *not* be exposed outside of run summary module
 /// The execution summary
 #[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
 pub struct ExecutionSummary<'a> {
     // this thread handles the state management
     #[serde(skip)]
     state_thread: tokio::task::JoinHandle<SummaryState>,
     #[serde(skip)]
     sender: mpsc::Sender<Message>,
     command: String,
-    package_inference_path: Option<&'a AnchoredSystemPath>,
-    started_at: DateTime<Local>,
+    #[serde(rename = "repoPath", skip_serializing_if = "Option::is_none")]
+    package_inference_root: Option<&'a AnchoredSystemPath>,
+    #[serde(serialize_with = "serialize_datetime")]
+    pub(crate) start_time: DateTime<Local>,
+    #[serde(serialize_with = "serialize_optional_datetime")]
+    pub(crate) end_time: Option<DateTime<Local>>,
+    pub(crate) exit_code: Option<u32>,
```

Comment:
I'm realizing I should have pushed back on this in the run summary PR. I don't think we want to be serializing this structure. This structure should be responsible on collecting the stats of individual runs. There should be another structure that matches the schema of the JSON/text we output that gets populated when it is requested. Basically rename this to`ExecutionSummaryTracker` and introduce a new `ExecutionSummary`

The Go code had poor boundaries between the display logic vs the collecting the data.
 - `end_time` and `exit_code` are only known once execution completes and will always be present so we could drop the wrapping `Option` if they got moved to the new `ExecutionSummary`
 - `package_inference_root` is only used for display
 - `start_time` gets passed in from the top only to be stored until the end for display

Reply:
Hmm yeah I'm starting to agree with that. I suspect what we should do is completely split out tracking and run summary. Because right now we're trying to do both at the same time (see `RunSummary` vs `RunSummaryInner`) and that just isn't working. We can make a bunch of tracking structs, then convert to summaries and print those.